### PR TITLE
REST endpoints to expose cluster/cache rebalancing

### PR DESCRIPTION
**Pull request by ryanemerson**
https://issues.redhat.com/browse/ISPN-11728
https://issues.redhat.com/browse/ISPN-11729
https://issues.redhat.com/browse/ISPN-13113

### CODE REVIEW

**Comment by tristantarrant** [comment]
Cli commands?

**Comment by ryanemerson** [comment]
> Cli commands?

I created ISPN-13113 for that, but  I can add it to this PR tomorrow :slightly_smiling_face:

**Review by gustavonalle — CHANGES_REQUESTED** [review]
LGTM, only missing API docs

**Comment by ryanemerson** [comment]
@tristantarrant Updated with CLI commands. Basic usage:

```
# Per cache
rebalance enable caches/mycache
rebalance disable caches/mycache

# Clusterwide
rebalance enable
rebalance disable
```

**Comment by ryanemerson** [comment]
> LGTM, only missing API docs

Docs added. I also reworked the REST resources so that `SecurityActions` were used where required.

File: server/rest/src/test/java/org/infinispan/rest/resources/CacheV2ResourceTest.java [hunk 1]
```diff
@@ -595,6 +598,36 @@ public void testGetProtoCacheConfig() {
...
+      ResponseAssertion.assertThat(response).isOk();
+      assertRebalancingStatus(cacheName, false);
+
+      response = join(cacheClient.enableRebalancing());
+      ResponseAssertion.assertThat(response).isOk();
+      assertRebalancingStatus(cacheName, true);
+   }
+
+   private void assertRebalancingStatus(String cacheName, boolean enabled) {
+      Security.doAs(ADMIN_USER, () -> {
```
Comment (danberindei) [inline server/rest/src/test/java/org/infinispan/rest/resources/CacheV2ResourceTest.java:617]: I'd rather use `TestingUtil.extractGlobalComponent()` and avoid explicit security code in the test.

**Review by danberindei — APPROVED** [review]
LGTM, one minor comment

File: server/rest/src/test/java/org/infinispan/rest/resources/CacheV2ResourceTest.java (hunk 1 above)
Comment (ryanemerson) [reply server/rest/src/test/java/org/infinispan/rest/resources/CacheV2ResourceTest.java]: Updated.

**Review by karesti — APPROVED** [review]

**Review by oraNod — APPROVED** [review]

**Comment by ryanemerson** [comment]
Rebased to add the docs suggestions to the relevant commits. I think we're good to go :slightly_smiling_face:

**Comment by karesti** [comment]
@ryanemerson I rebased, if the build is good I will merge

**Pull request merged**